lib/p2p/P2PRepository.ts

```diff
@@ -19,11 +19,19 @@ class P2PRepository {
   }
 
   public addHost = async (host: db.HostFactory): Promise<db.HostInstance> => {
-    return this.models.Host.upsert(<db.HostAttributes>host , { returning: true })[0];
+    return this.models.Host.findOne({ where: { address: host.address, port: host.port } }).then((result): Promise<db.HostInstance> => {
+      if (result) {
+        return Promise.resolve<db.HostInstance>(result.update(<db.HostAttributes>host));
+      } else {
+        return Promise.resolve<db.HostInstance>(this.models.Host.create(<db.HostAttributes>host));
+      }
+    });
   }
 
-  public addHosts = async (hosts: db.HostFactory[]): Promise<db.HostInstance[]> => {
-    return this.models.Host.bulkCreate(<db.HostAttributes[]>hosts);
+  public addHosts = async (hosts: db.HostFactory[]): Promise<void> => {
+    hosts.forEach((host) => {
```

Comment:
I figure we should also be able to leave this as is - although we're not using `addHosts` anywhere in the codebase currently.

Reply:
@sangaman i am using this in db init gulp task to insert hosts in to the database.

Reply:
Ah oops I forgot to check the tests. I still figure we can leave this as before though. Or you can pass the `ignoreDuplicates` option to `bulkCreate`: http://docs.sequelizejs.com/class/lib/model.js~Model.html#static-method-bulkCreate.

Reply:
@sangaman tried that but i couldn't make work though!

Reply:
What problem were you getting? Uniqueness constraints? You can try the `updateOnDuplicate` option too.

It's not a big deal for tasks and tests and stuff really, but I definitely think we shouldn't be hitting the database to check if hosts exist during regular operation.